syntax = "proto2";

message Query {
    required int32 command = 1;
    repeated Filter filtersList = 2;
    repeated ValueSetting settingsList = 3;

    message Filter {
        repeated Comparator compList = 1;
    }

    message ValueSetting {
        required KeyValuePair kv = 1;
    }

    message Comparator {
        required int32 operation = 1;
        required KeyValuePair kv = 2;
    }

    message KeyValuePair {
        required string key = 1;
        required int32 valueType = 2;
        required int64 valueInt = 3;
        required float valueReal = 4;
        required string valueString = 5;
    }
}

message Response {
    required int32 last = 2;
    required string rString = 1;
}